cli/provision.py

```diff
@@ -44,13 +50,13 @@ def process_topology_args(clg_args, app_settings_dir):
                     "Topology node should be in format  <number>_<node role>. "
                     "Current value: '{}' ".format(topology_item))
             # todo(obaraov): consider moving topology to config on constant.
-            node_file = os.path.join(
-                app_settings_dir, 'topology', node_type + '.yaml')
-            with open(node_file) as stream:
-                topology_dict[node_type] = yaml.load(stream)
-                topology_dict[node_type]['amount'] = int(number)
+            topology_dict[node_type] = spec.find_file(node_type + ".yaml",
+                                                      topology_dir)
+            topology_dict[node_type]['amount'] = int(number)
+
+        self.value = topology_dict
 
-        clg_args['topology'] = topology_dict
+clg.TYPES.update({'Topology': TopologyArgument})
```

Comment:
I'd suggest to have  types update somewhere in one place. Now we have update in provision.py and in the spec.py file.